javascript/src/URDFClasses.js

```diff
@@ -242,9 +246,81 @@ class URDFJoint extends URDFBase {
             }
 
             case 'floating':
+                // no-op if all values are identical to existing value or are null
+                if (this.jointValue.every((value, index) => values[index] === value || values[index] === null)) return didUpdate;
+                // anonymous block scope to prevent variable name clobbering with other cases in the switch
+                {
+                    // Floating joints have six degrees of freedom: X, Y, Z, R, P, Y.
+                    const [posX, posY, posZ, roll, pitch, yaw] = values;
+                    this.position.copy(this.origPosition);
+                    // Respect origin RPY when setting position
+                    if (posX !== this.jointValue[0] && posX !== null) {
+                        _tempAxis.set(1, 0, 0).applyEuler(this.rotation);
+                        this.position.addScaledVector(_tempAxis, posX);
+                        didUpdate = true;
+                    }
+                    if (posY !== this.jointValue[1] && posY !== null) {
+                        _tempAxis.set(0, 1, 0).applyEuler(this.rotation);
```

Comment:
We'll want to make sure we use the "origQuaternion" here so we're operating in the frame defined by the original joint state otherwise the rotation of these axes will be dependent on the previous state of the rotation based on prior joint angles.

Reply:
The prismatic joint will never have the rotation changed so the orientation in`this.rotation` will never be different from `this.origQuaternion` in that case.